image/image_linux.go

```diff
@@ -569,6 +576,15 @@ func (s *imageSeeder) downloadSnaps(snapsToDownload []*seedwriter.SeedSnap, curS
 	if err != nil {
 		return nil, err
 	}
+
+	for _, sn := range snapsToDownload {
+		dc := downloadedSnaps[sn.SnapName()].Components
+		for i := range sn.Components {
+			sn.Components[i].Info = dc[i].Info
```

Comment:
isn't this set by SetInfo above already when we pass the cinfos? or the value different here? if it is, when does the Writer get a chance to check it?

Reply:
Right, this is actually not needed. I've moved also setting the path to SetInfo so now things are done similarly to the snaps case.